### PR TITLE
Badge: Add ReacNode support for count prop

### LINKED ISSUE #11134: Typescript error when passing a react node to badge count
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.6.5

### Environment
Chrome

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/61qwzlwlrk)

### Steps to reproduce
Pass a react node to badge's `count` prop:
```
<Badge count={<Icon type="clock-circle-o" />} />
```

Will result in a typescript error:
```
Types of property 'count' are incompatible.
Type 'Element' is not assignable to type 'string | number | undefined'.
```

### What is expected?
We should be able to pass in a react node to the badge's `count` while using typescript as document said.

### What is actually happening?
Typescript will throw an error since the type of badge's `count` does not accept a react node. It currently accepts only `string | number | undefined` for badge's count.

### PR DESCRIPTION
**Pull request by supra28**
Fixes [issue](https://github.com/ant-design/ant-design/issues/11134)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for branch `feature`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

### CODE REVIEW

**Comment by supra28** [comment]
Is the CI broken?

**Comment by supra28** [comment]
rebased after snapshot updates

**Comment by zombieJ** [comment]
It's better to add a demo about it : )

File: components/badge/ScrollNumber.tsx [hunk 1]
```diff
@@ -116,8 +117,12 @@ export default class ScrollNumber extends Component<ScrollNumberProps, ScrollNum
   }
 
   renderNumberElement() {
+    const props = this.props;
     const state = this.state;
-    if (!state.count || isNaN(state.count as number)) {
+    if(props.displayComponent){
```
Comment (afc163) [inline components/badge/ScrollNumber.tsx]: code style is a mess...

Comment (supra28) [reply components/badge/ScrollNumber.tsx]: What exactly should I change here? It was already written this way.
This is my first time, pls give a few pointers.

Comment (afc163) [reply components/badge/ScrollNumber.tsx]:
```diff
- if(props.displayComponent){
+ if (props.displayComponent) {
```

File: components/badge/index.tsx [hunk 2]
```diff
@@ -105,7 +111,8 @@ export default class Badge extends React.Component<BadgeProps, any> {
         data-show={!hidden}
         className={scrollNumberCls}
         count={displayCount}
-        title={title || count}
+        displayComponent = {displayComponent}
```
Comment (afc163) [inline components/badge/index.tsx]:
```diff
- displayComponent = {displayComponent}
+ displayComponent={displayComponent}
```

File: components/badge/ScrollNumber.tsx [hunk 3]
```diff
@@ -116,8 +117,12 @@ export default class ScrollNumber extends Component<ScrollNumberProps, ScrollNum
...
+      return props.displayComponent
```
Comment (afc163) [inline components/badge/ScrollNumber.tsx]: `;`

File: components/badge/ScrollNumber.tsx [hunk 4]
```diff
@@ -116,8 +117,12 @@ export default class ScrollNumber extends Component<ScrollNumberProps, ScrollNum
   }
 
   renderNumberElement() {
+    const props = this.props;
```
Comment (afc163) [inline components/badge/ScrollNumber.tsx]:
```diff
- const props = this.props;
+ const { displayComponent } = this.props;
```

**Comment by afc163** [comment]
Could you add a test case for this feature?

**Comment by supra28** [comment]
@afc163 Yeah i'll make changes and add a test case this week, as soon as I get some time.

**Comment by afc163** [comment]
@supra28

**Comment by afc163** [comment]
Merged in https://github.com/ant-design/ant-design/commit/fc6ac421c26b53e23b6fa09ef98ace1428cc67f3, thx for sending PR

**Pull request closed without merging**